modules/systemd-journal/tests/test_systemd_journal.c

```diff
@@ -487,21 +511,24 @@ _test_program_field_test(TestCase *self, TestSource *src, LogMessage *msg)
   journald_get_cursor(journal, &cursor);
   if (strcmp(cursor, "no SYSLOG_IDENTIFIER") != 0)
     {
-      assert_string(log_msg_get_value(msg, LM_V_PROGRAM, NULL), "syslog_program", ASSERTION_ERROR("Bad program name"));
+      cr_assert_str_eq(log_msg_get_value(msg, LM_V_PROGRAM, NULL), "syslog_program", "%s", "Bad program name");
       g_free(cursor);
     }
   else
     {
-      assert_string(log_msg_get_value(msg, LM_V_PROGRAM, NULL), "comm_program", ASSERTION_ERROR("Bad program name"));
+      cr_assert_str_eq(log_msg_get_value(msg, LM_V_PROGRAM, NULL), "comm_program", "%s", "Bad program name");
       g_free(cursor);
       test_source_finish_tc(src);
     }
 }
 
-void
-test_journal_reader(void)
+Test(systemd_journal, test_journal_reader)
 {
-  TestSource *src = test_source_new(configuration);
+  const gchar *persist_file = "test_systemd_journal1.persist";
+
+  _init_cfg_with_persist_file(persist_file);
```

Comment:
As far as I can see all of your tests starts with this functions, it is just preparing common things for the tests. I would use this and the deinit for `init` and `fini`.

Reply:
@Kokan : I want a different file name for each test case. So, I guess putting this function in init and fini will give me same file name everytime.

Reply:
Yes, still some part could be moved there; or perhaps we could have some different api creating persist files with generated names for UT.

Reply:
@Kokan : So, do I need to change the code with your suggestion?

Reply:
I am fine with it like this (see approve from my side), but sure would prefer some more cleanup.